code/controllers/master.dm

```diff
@@ -43,37 +50,51 @@ var/global/datum/controller/master/Master = new()
 	// Tell qdel() to Del() this object.
 	return QDEL_HINT_HARDDEL_NOW
 
+//returns 1 if we created a new mc, 0 if we couldn't due to a recent restart,
+//	-1 if we encountered a runtime trying to recreate it
+/proc/Recreate_MC()
+	if (world.time > MC_restart_timeout)
+		return 0
+	. = -1 //so if we runtime, things know we failed
+	MC_restart_timeout = world.time + (50 * ++MC_restart_count)
+	Master.processing = -1 //stop ticking this one
+	try
+		new/datum/controller/master()
```

Comment:
`MC_restart_timeout = world.time + (50 * ++MC_restart_count)` is designed to catch this and throttle the attempts to recreate the mc